writer/proto/proto.go

```diff
@@ -64,17 +70,40 @@ func WriteServiceToProto(ps *parser.ParsedService, outputDir string) ([]byte, er
 	sb.SetComments(builder.Comments{
 		LeadingComment: "A service.",
 	})
+
+	// Add resources to MessageStorage.
+	err := GenerateResourceMessages(ps.ResourceByType, ps.ObjectByType, ps, m)
+	if(err != nil) {
+		return nil, err
+	}
+
 	for _, r := range getSortedResources(ps.ResourceByType) {
-		err := AddResource(r, fb, sb)
+		err := AddResource(r, ps, fb, sb, m)
 		if err != nil {
 			return []byte{}, fmt.Errorf("adding resource %v failed: %w", r.Kind, err)
 		}
 	}
+	for _, r := range getSortedResources(ps.ObjectByType) {
+		err := AddResource(r, ps, fb, sb, m)
+		if err != nil {
+			return []byte{}, fmt.Errorf("adding object %v failed: %w", r.Kind, err)
+		}
+	}
 	fb.AddService(sb)
 	fd, err := fb.Build()
 	if err != nil {
 		return []byte{}, fmt.Errorf("unable to build service file %v: %w", fb.GetName(), err)
 	}
+
+	// protoreflect sometimes adds "import {generated-file-0001}.proto" unnecessarily.
```

Comment:
nit: might be worth filing a bug on this (with protoreflect)

Reply:
Yeah, I should...

I'll try to get around to making a cleaner repro of this, but that'll be later...